src/shared/user-record-show.c

```diff
@@ -132,7 +132,23 @@ void user_record_show(UserRecord *hr, bool show_full_group_info) {
                         break;
                 }
 
-                printf(" Password OK: %syes%s\n", ansi_highlight_green(), ansi_normal());
+                if (strv_isempty(hr->hashed_password)) {
+                        printf(" Password OK: %sno%s (none set)\n", ansi_highlight(), ansi_normal());
```

Comment:
one more check would be good here: if you are unprivileged, and inquire the record of another user (i.e. not of yourself) we'll suppress the hashed password info (this takes inspiration from Linux /etc/shadow being inaccessible to regular users). Thus, if strv_isempty(hashed_password) holds, this could mean two things: that there really are no passwords set, or that we simply don't know about them because of this privileges issue. Thankfully, there's the `hr->incomplete` boolean we can use to distinguish these two cases: if it's true then we know the data is incomplete. Thus, if the field is empty *and* that boolean set we probably should supress any output about the pw here, we have to assume we simply don't know. Thus, immediately before this printf(), please add:

```c
if (hr->incomplete) /* Record might be incomplete, due to privs */
        break;
```

Reply:
You included the check, but without the comment. Please add the comment too, it's rather helpful.